agent/testflinger_agent/tests/test_job.py

```diff
@@ -173,6 +175,111 @@ def test_run_test_phase_with_run_exception(
         assert exit_event == "setup_fail"
         assert exit_reason == "failed"
 
+    def run_testcmds(self, job_data, client, tmp_path) -> str:
+        # create the outcome file manually
+        with open(tmp_path / "testflinger-outcome.json", "w") as outcome_file:
+            outcome_file.write("{}")
+        # create the agent configuration file manually
+        with open(tmp_path / "default.yaml", "w") as config_file:
+            config_file.write("agent_name: agent-007")
+        # write the job data to `testflinger.json` manually:
+        # (so that the device connector can pick it up)
+        with open(tmp_path / "testflinger.json", "w") as job_file:
+            json.dump(job_data, job_file)
+
+        # running the device connector will result in running the `test_cmds`
+        self.config["test_command"] = (
+            "testflinger-device-connector fake_connector runtest "
+            f"--config {tmp_path}/default.yaml "
+            f"{tmp_path}/testflinger.json"
+        )
+
+        # run the test phase of the job
+        job = _TestflingerJob(job_data, client)
+        exit_code, _, _ = job.run_test_phase(TestPhase.TEST, tmp_path)
+        assert exit_code == 0
+
+        # capture the output
+        with open(tmp_path / "test.log") as log:
+            return log.read()
+
+    def test_run_test_phase_sensitive(self, client, tmp_path, requests_mock):
+        requests_mock.post(rmock.ANY, status_code=200)
+
+        # sensitive patterns are added to the agent configuration file:
+        # this will result in the use of a masking command runner
+        self.config["sensitive_patterns"] = ["[Cc][a-z]*n"]
+
+        # create the job data
+        job_data = {
+            "test_data": {"test_cmds": "echo -n Message: Captain Major"}
+        }
+
+        log_data = self.run_testcmds(job_data, client, tmp_path)
+        print(log_data)
```

Comment:
are you supposed to print log data to stdout, not stderr?

Reply:
This is a test, so neither `stdout` nor `stderr` are printed when the test is successful. They are however both captured and printed (separately) when the test fails, so this `print` is meant as a diagnostic, i.e. you only see it when the test fails and might help you understand what went wrong. I'd happily change this to print to `stderr` if you feel it makes any difference.